lib/matplotlib/table.py

```diff
@@ -453,19 +476,28 @@ def _auto_set_font_size(self, renderer):
 
         if len(self._cells) == 0:
             return
-        fontsize = list(six.itervalues(self._cells))[0].get_fontsize()
-        cells = []
+
+        fontsize = None
```

Comment:
It would be simpler to start this as `fontsize = self.FONTSIZE` and set `cell.set_fontsize(fontsize)` at the top of the loop.  I think this would eliminate all of the null testing.